File: #Leetcode/Longest_Common_Prefix.py
# Question:-

# Write a function to find the longest common prefix string amongst an array of strings.
# If there is no common prefix, return an empty string "".

# Solution:-

from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        prefix = strs[0]

        for s in strs[1:]:
            while not s.startswith(prefix):
                prefix = prefix[:-1]
                if not prefix:
                    return "none"

        return prefix
    
t = int(input("Enter no. of test cases: "))
for i in range (0,t):
    strs = input("Enter a list of strings (comma-separated): ").split(",")

    solution = Solution()
    result = solution.longestCommonPrefix(strs)
    print(result)

File: #Leetcode(C++)/Longest_Common_Prefix.cpp
// #Question:-

// Write a function to find the longest common prefix string amongst an array of strings.
// If there is no common prefix, return an empty string "".

// Solution:-

#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& v) {
        string ans="";
        sort(v.begin(),v.end());
        int n=v.size();
        string first=v[0],last=v[n-1];
        for(int i=0;i<min(first.size(),last.size());i++){
            if(first[i]!=last[i]){
                return ans;
            }
            ans+=first[i];
        }
        return ans;
    }
};

int main(){
        int t;
        cout << "Enter the no. of test cases: ";
        cin >> t;
        while(t--){
            int n;
            cout << "Enter the number of strings: ";
            cin >> n;
            vector<string> v(n);
            cout << "Enter the strings: " << endl;
            for (int i = 0; i < n; i++) {
                cin >> v[i];
            }
            Solution sol;
            string result = sol.longestCommonPrefix(v);
            cout << "Longest Common Prefix: " << result << endl;
        }
    return 0;
}